PlanPluginBridge/src/main/java/com/djrapitops/pluginbridge/plan/nucleus/NucleusData.java

```diff
@@ -0,0 +1,311 @@
+                    String warner = "Unknown";
+                    if (warning.getWarner().isPresent()) {
+                        User warnerUser = getUser(warning.getWarner().get());
+                        if (warnerUser != null) {
+                            warner = warnerUser.getName();
+                        }
+                    }
+
+                    warningsTable.addRow(warner, warning.getReason());
+                }
+
+                inspectContainer.addTable("Warnings", warningsTable);
+            }
+        }
+
+        return inspectContainer;
+    }
+
+    @Override
+    public AnalysisContainer getServerData(Collection<UUID> uuids, AnalysisContainer analysisContainer) {
+        if (NucleusAPI.getWarpService().isPresent()) {
+            NucleusWarpService warpService = NucleusAPI.getWarpService().get();
+
+            List<Warp> warps = warpService.getAllWarps();
+            analysisContainer.addValue(getWithIcon("Warp count", Icon.called("map-marker-alt").of(Color.BLUE)), warps.size());
+
+            if (!warps.isEmpty()) {
+                TableContainer warpsTable = new TableContainer(
+                    getWithIcon("Name", Icon.called("map-marker-alt").of(Family.SOLID)),
+                    getWithIcon("Description", Icon.called("sticky-note").of(Family.REGULAR)),
+                    getWithIcon("Category", Icon.called("list").of(Family.SOLID))
+                );
+
+                for (Warp warp : warps) {
+                    String description = "None";
+                    String category = "None";
+
+                    if (warp.getDescription().isPresent()) {
+                        description = HtmlUtils.swapColorsToSpan(TextSerializers.FORMATTING_CODE.serialize(warp.getDescription().get()));
+                    }
+
+                    if (warp.getCategory().isPresent()) {
+                        category = warp.getCategory().get();
+                    }
+
+                    warpsTable.addRow(warp.getName(), description, category);
+                }
+
+                analysisContainer.addTable("Warps", warpsTable);
+            }
+        }
+
+        if (NucleusAPI.getJailService().isPresent()) {
+            NucleusJailService jailService = NucleusAPI.getJailService().get();
+
+            Map<String, NamedLocation> jails = jailService.getJails();
+            analysisContainer.addValue(getWithIcon("Jail count", Icon.called("bars").of(Family.SOLID).of(Color.TEAL)), jails.size());
+
+            if (!jailService.getJails().isEmpty()) {
+                TableContainer jailsTable = new TableContainer(getWithIcon("Jail", Icon.called("bars").of(Family.SOLID)));
+
+                for (Map.Entry<String, NamedLocation> jail : jails.entrySet()) {
+                    jailsTable.addRow(jail.getKey());
+                }
+
+                analysisContainer.addTable("Jails", jailsTable);
+            }
+        }
+
+        if (NucleusAPI.getKitService().isPresent()) {
```

Comment:
This if block could be refactored into it's own method
`ifPresent(service -> newMethod(service, inspectContainer))`